roms_tools/utils.py

```diff
@@ -0,0 +1,172 @@
+from numbers import Integral
+
+import numpy as np
+import xarray as xr
+
+
+def partition(
+    ds: xr.Dataset, nx: int = 1, ny: int = 1
+) -> tuple[list[int], list[xr.Dataset]]:
+    """
+    Split a ROMS dataset up into nx by ny spatial tiles.
+    """
+
+    # TODO also check they are positive integers
+    if not isinstance(nx, Integral) or not isinstance(ny, Integral):
+        raise ValueError("nx and ny must be integers")
+
+    # 'eta_rho' and 'xi_rho' are always expected to be present
+    partitionable_dims_maybe_present = ["eta_v", "xi_u", "eta_coarse", "xi_coarse"]
+    dims_to_partition = ["eta_rho", "xi_rho"] + [
+        d for d in partitionable_dims_maybe_present if d in ds.dims
+    ]
+
+    # if eta is periodic there are no ghost cells along those dimensions
+    if "eta_v" in ds.sizes and ds.sizes["eta_rho"] == ds.sizes["eta_v"]:
+        # TODO how are we supposed to know if eta is periodic if eta_v doesn't appear? partit.F doesn't say...
+        n_eta_ghost_cells = 0
+    else:
+        n_eta_ghost_cells = 1
+
+    # if xi is periodic there are no ghost cells along those dimensions
+    if "xi_u" in ds.sizes and ds.sizes["xi_rho"] == ds.sizes["xi_u"]:
+        n_xi_ghost_cells = 0
+    else:
+        n_xi_ghost_cells = 1
+
+    def integer_division_or_raise(a: int, b: int) -> int:
+        remainder = a % b
+        if remainder == 0:
+            return a // b
+        else:
+            raise ValueError(
```

Comment:
Maybe it's worth including a test that checks that this ValueError is raised.

Reply:
Done in https://github.com/CWorthy-ocean/roms-tools/pull/101/commits/f57788020d7539a3367bf70d8259e9ca9ea6860d